src/Uno.UI/UI/Xaml/Controls/Canvas/Canvas.Layout.cs

```diff
@@ -13,19 +13,9 @@ public partial class Canvas : ICustomClippingElement
 	{
 		protected override Size MeasureOverride(Size availableSize)
 		{
+			// A canvas does not have dimensions and will always return zero even with a chidren collection.
 			double maxWidth = 0, maxHeight = 0;
 
-			foreach (var child in Children.Where(c => c is DependencyObject))
-			{
-				var childX = GetLeft(child as DependencyObject);
-				var childY = GetTop(child as DependencyObject);
-
-				var measuredSize = MeasureElement(child, new Size(double.PositiveInfinity, double.PositiveInfinity));
-
-				maxHeight = Math.Max(maxHeight, measuredSize.Height + childY);
-				maxWidth = Math.Max(maxWidth, measuredSize.Width + childX);
-			}
-
 			return new Size(maxWidth, maxHeight);
```

Comment:
COuld just return `new Size(0, 0)` or even `default`.

Reply:
True, was just to remember about what size with talked about.

Reply:
And i'm finishing some testing on internal project that use canvas to be sure we really don't need to put extra condition ^^